app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -457,7 +457,16 @@ function isBoltOnlyPayment()
     {
         return Mage::getStoreConfig('payment/boltpay/skip_payment');
     }
-
+    
+    /**
+     * Return disabled customer groups for Bolt
+     * @return array
+     */
+    public function disableCustomerGroups()
```

Comment:
When creating a "getter" method, by convention in M1, the method names start with `get`.  Therefore this 
 should be named `getDisabledCustomerGroups`.

However, considering what you are trying to do and the comment below about corrupting/overriding a well defined method, it is better that you create a `private` method here and call it `isCustomerGroupAllowed` and returning a boolean true|false.  This method should then be called and `&&` to this condition:

**https://github.com/BoltApp/bolt-magento1/blob/master/app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php#L664** in the `isAllowedConnectJsOnCurrentPage` method of this same class.

Reply:
@mccottry add new condition in the method isAllowedConnectJsOnCurrentPage - bad practice.
Now in this method we call independently two methods. First method could return true and really we don't need second method. This is on one hand. Just for performance it's better create chain of rules. If first rule returns false (Bolt is not allowed) we don't need do any other calculations. Because of this i added method isCustomerGroupDisabled in the method isAllowedOnCurrentPageByRoute
On the other hand this is not a problem for boolean logic get result based on two parameters, but much more complex get result based on three parameters.